Digipost/src/main/java/no/digipost/android/gui/metadata/ExternalLinkWebview.java

```diff
@@ -0,0 +1,233 @@
+
+    @Override
+    protected void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_externallink_webview);
+        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
+        setSupportActionBar(toolbar);
+        final ActionBar actionBar = getSupportActionBar();
+        if (actionBar != null) {
+            actionBar.setDisplayHomeAsUpEnabled(true);
+            actionBar.setBackgroundDrawable(new ColorDrawable(0xff454545));
+            if (android.os.Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+                Window window = this.getWindow();
+                window.clearFlags(WindowManager.LayoutParams.FLAG_TRANSLUCENT_STATUS);
+                window.addFlags(WindowManager.LayoutParams.FLAG_DRAWS_SYSTEM_BAR_BACKGROUNDS);
+                window.setStatusBarColor(ContextCompat.getColor(this, R.color.metadata_externalbrowser_top_background));
+            }
+        }
+
+        webView = (WebView) findViewById(R.id.externallink_webview);
+        webView.getSettings().setJavaScriptEnabled(true);
+        webView.getSettings().setDomStorageEnabled(true);
+        Bundle bundle = getIntent().getExtras();
+        fileUrl = bundle.getString("url", "https://www.digipost.no");
+
+        enableCookies(webView);
+        webView.setWebViewClient(new WebViewClient() {
+            @Override
+            public void onPageFinished(WebView view, String url) {
+                super.onPageFinished(view, url);
+                if(actionBar != null) {
+                    actionBar.setTitle(view.getTitle());
+                    actionBar.setSubtitle(view.getUrl());
+                }
+            }
+        });
+
+        webView.setDownloadListener(new DownloadListener() {
+            @Override
+            public void onDownloadStart(String url, String userAgent, String content, String mimeType, long contentLength) {
+                fileName = URLUtil.guessFileName(url, content, mimeType);
+                fileUrl = url;
+                onComplete = new BroadcastReceiver() {
+                    @Override
+                    public void onReceive(Context context, Intent intent) {
+                        showDownloadSuccessDialog(context);
+                    }
+                };
+
+                registerReceiver(onComplete, new IntentFilter(DownloadManager.ACTION_DOWNLOAD_COMPLETE));
+                if (!mimeType.equals("text/html")) {
+                    if(FileUtilities.isStorageWriteAllowed(getApplicationContext())) {
+                        showDownloadDialog(userAgent, content, mimeType, contentLength);
+                    }else{
+                        showMissingPermissionsDialog();
+                    }
+                }
+            }
+        });
+
+        if(FileUtilities.isStorageWriteAllowed(this)){
+            webView.loadUrl(fileUrl);
+        }else{
+            showPermissionsDialog();
+        }
```

Comment:
Du har generelt veldig mange statements per funksjon.
Går det an å refaktorere "oppgaver" ut i funksjoner, med konsise deskriptive navn? Slik at koden kan bli mindre kompleks, og mer lett forståelig. Kanskje formålet bak kode-inndelingene blir mer synlig for både nestemann, og også deg?